File: Frontend/src/app/_models/logged-equivalent-course.ts
import { GUID } from 'src/utils/guid';
import { ExemptedCourse } from './exempted-course';

export interface LoggedEquivalentCourse {
  id: GUID;
  exemptedCourse: ExemptedCourse;
  hostCourseCode: string;
  hostCourseName: string;
  logTime?: Date;
  hostCourseEcts: number;
  hostSchool: string;
}

File: Backend/Entities/LoggedEquivalentCourse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Backend.Entities
{
    // Accepted equivalence course pairs
    public class LoggedEquivalentCourse : LoggedCourse
    {
        public ExemptedCourse ExemptedCourse { get; set; }
        public string HostCourseCode { get; set; }
        public string HostCourseName { get; set; }
        public double HostCourseECTS { get; set; }
    }
}
